presto-clickhouse/src/main/java/io/prestosql/plugin/clickhouse/ClickHouseMetadata.java

```diff
@@ -0,0 +1,234 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.clickhouse;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import io.airlift.log.Logger;
+import io.airlift.slice.Slice;
+import io.prestosql.plugin.jdbc.JdbcColumnHandle;
+import io.prestosql.plugin.jdbc.JdbcIdentity;
+import io.prestosql.plugin.jdbc.JdbcMetadata;
+import io.prestosql.plugin.jdbc.JdbcMetadataConfig;
+import io.prestosql.plugin.jdbc.JdbcOutputTableHandle;
+import io.prestosql.plugin.jdbc.JdbcTableHandle;
+import io.prestosql.spi.connector.ColumnHandle;
+import io.prestosql.spi.connector.ColumnMetadata;
+import io.prestosql.spi.connector.ConnectorInsertTableHandle;
+import io.prestosql.spi.connector.ConnectorNewTableLayout;
+import io.prestosql.spi.connector.ConnectorOutputMetadata;
+import io.prestosql.spi.connector.ConnectorSession;
+import io.prestosql.spi.connector.ConnectorTableHandle;
+import io.prestosql.spi.connector.ConnectorTableMetadata;
+import io.prestosql.spi.connector.SchemaTableName;
+import io.prestosql.spi.connector.SchemaTablePrefix;
+import io.prestosql.spi.connector.TableNotFoundException;
+import io.prestosql.spi.statistics.ComputedStatistics;
+import io.prestosql.spi.type.Type;
+
+import javax.inject.Inject;
+
+import java.util.Collection;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static java.lang.String.format;
+import static java.lang.String.join;
+import static java.util.Objects.requireNonNull;
+
+public class ClickHouseMetadata
+        extends JdbcMetadata
```

Comment:
Why can't we use JdbcMetadata itself ?

Reply:
I need extend `JdbcMetadata` to implement itself `createTable` function .
I want append some table properties like `engine = log` to normal create table sql.
Is there exists simple way to resolve above ?

Reply:
We can override the `BaseJdbcClient#createTable` where we can fetch table properties from `ConnectorTableMetadata`

Reply:
thanks, I will try it

Reply:
> We can override the `BaseJdbcClient#createTable` where we can fetch table properties from `ConnectorTableMetadata`

hi @Praveen2112  

I try to define table properties in `ClickHouseTableProperties.java`  and as override method `createTable` in `ClickHouseClient.java` , I try get table's properties in the function .

when I  try to create table like the following:

```sql
create table p_tbl(id int) with (engine='Log')
```
the following errors occurred:

````
io.prestosql.spi.PrestoException: Catalog 'clickhouse' does not support table property 'engine'
	at io.prestosql.metadata.AbstractPropertyManager.getProperties(AbstractPropertyManager.java:96)
	at io.prestosql.execution.CreateTableTask.internalExecute(CreateTableTask.java:190)
	at io.prestosql.execution.CreateTableTask.execute(CreateTableTask.java:87)
	at io.prestosql.execution.CreateTableTask.execute(CreateTableTask.java:69)
	at io.prestosql.execution.DataDefinitionExecution.start(DataDefinitionExecution.java:167)
	at io.prestosql.execution.SqlQueryManager.createQuery(SqlQueryManager.java:245)
	at io.prestosql.dispatcher.LocalDispatchQuery.lambda$startExecution$7(LocalDispatchQuery.java:132)
	at io.prestosql.$gen.Presto_339_19_ged4dfb3____20200804_145539_2.run(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630)
	at java.base/java.lang.Thread.run(Thread.java:832)
```

I also find in other connector (such as `phoenix-connector`) , it uses `PhoenixMetadata.java` to process table's properties, is this normal way to do it ?